### DISCUSSION TITLE
Advice for managing chezmoi data?

### DISCUSSION THREAD

**Discussion by kartikynwa** [twpayne/chezmoi#2114] [Q&A] [answered]
I have just started migrating my dotfiles repository from disastrous hack job to chezmoi. I am currently experimenting with templating and it looks like something that would have helped me a lot in the past.

Now I am trying to figure out what the best way to keep track of chezmoi data would be. My use case as of now is very simple. An example is the names of Xrander sinks. In `.chezmoidata.yaml`, I have:

```yaml
i3:
  laptop_monitor: DP-0
  external_monitor: DP-2
```

This file is committed to the repository.

If I have to change these values on my work laptop, I can create `~/.config/chezmoi/chezmoi.yaml` and change the values there.

Is there a better way than this?

My main concern is that if I am setting up a new machine, I don't want to have to figure out which data values to provide by either reading through the templates or via trial-and-error.

Keeping this in mind, I can commit `.chezmoidata.yaml` to the repository as long as it does not have sensitive information, then I can use this as a reference to populate the data field in the chezmoi config file.

If I am making a mistake somewhere or if you can suggest any improvements, feel free to let me know.

**Comment by twpayne** [comment, marked as answer]
As a rough guideline:

* `~/.config/chezmoi/chezmoi.yaml` is for data which is specific to the current machine.
* `~/.local/share/chezmoi/.chezmoidata.yaml` is for data which is shared across all of your machines.

Personally, use a [configuration file template](https://github.com/twpayne/dotfiles/blob/master/home/.chezmoi.toml.tmpl) to generate my machine-specific configuration file.

**Reply by danielo515** [reply]
I think this should made more clear in the documentation.
The documentation (and some issues I read) is very insistent in the fact  that you should not include chezmoi.yaml as part of your source, that it is machine specific, etc, etc. But I think the point of having cross-machine management is, precisely, to be able to configure everything, including chezmoi itself, based on the machine.
Mentioning that, if you want to configure chezmoi using chezmoi itself you should probably use a template is something that will clarify a lot of user's doubts (at least users like me).
Also, I think this is the first time I see the chezmoidata.yaml file, and I only found it in the docs by searching for it. Is it mentioned in some other guide?

**Reply by twpayne** [reply]
> The documentation (and some issues I read) is very insistent in the fact that you should not include chezmoi.yaml as part of your source, that it is machine specific, etc, etc. But I think the point of having cross-machine management is, precisely, to be able to configure everything, including chezmoi itself, based on the machine.
> Mentioning that, if you want to configure chezmoi using chezmoi itself you should probably use a template is something that will clarify a lot of user's doubts (at least users like me).

This is explicitly mentioned in [the setup user guide](https://www.chezmoi.io/user-guide/setup/#create-a-config-file-on-a-new-machine-automatically). Do you have a specific suggestion for how the documentation should be changed?

> Also, I think this is the first time I see the chezmoidata.yaml file, and I only found it in the docs by searching for it. Is it mentioned in some other guide?

Yes, it's in the [templating guide](https://www.chezmoi.io/user-guide/templating/#template-data).

**Reply by danielo515** [reply]
Yes, that's quite clear. I don't know how I didn't saw it earlier.
Thanks

**Reply by kartikynwa** [reply]
Finally managed to figure out a decent way to configure looking at your dotfiles for inspiration: https://github.com/kartikynwa/dotfiles/blob/main/dotfiles/dot_config/chezmoi/chezmoi.toml.tmpl

Thanks very much.

**Reply by twpayne** [reply]
Great! Would you mind marking this as the answer?

**Reply by kartikynwa** [reply]
I think for "General" type discussions GitHub does not allow picking an answer.

**Reply by twpayne** [reply]
Thanks, I've moved it to Q&A now.
